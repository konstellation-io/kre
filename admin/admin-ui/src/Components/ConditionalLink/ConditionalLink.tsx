import React, { ReactElement, FunctionComponent } from 'react';
import { Link } from 'react-router-dom';

type Props = {
  children: ReactElement;
  to: string;
  LinkType?: typeof Link;
  linkProps?: object;
  disabled?: boolean;
};
const ConditionalLink: FunctionComponent<Props> = ({
  LinkType = Link,
  children,
  to = '',
  disabled = false,
  linkProps = {}
}) => {
  return !disabled ? (
    <LinkType to={to} {...linkProps}>
      {children}
    </LinkType>
  ) : (
    children
  );
};

export default ConditionalLink;
