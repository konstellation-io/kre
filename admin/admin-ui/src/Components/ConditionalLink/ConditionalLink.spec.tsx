import React from 'react';
import ConditionalLink from './ConditionalLink';
import { Link, NavLink } from 'react-router-dom';
import { shallow } from 'enzyme';

describe('ConditionalLink', () => {
  const ROUTE = '/some-route/index';
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ConditionalLink to={ROUTE}>
        <div>Sample text</div>
      </ConditionalLink>
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('shows right default values', () => {
    expect(wrapper.exists(Link)).toBeTruthy();
    expect(wrapper.text()).toBe('Sample text');
    expect(wrapper.find(Link).prop('to')).toBe(ROUTE);
  });

  it('shows right link type', () => {
    wrapper.setProps({ LinkType: NavLink });

    expect(wrapper.exists(NavLink)).toBeTruthy();
    expect(wrapper.text()).toBe('Sample text');
    expect(wrapper.find(NavLink).prop('to')).toBe(ROUTE);
  });

  it('do not show a link when disabled', () => {
    wrapper.setProps({ disabled: true });

    expect(wrapper.exists(Link)).toBeFalsy();
    expect(wrapper.text()).toBe('Sample text');
  });
});
