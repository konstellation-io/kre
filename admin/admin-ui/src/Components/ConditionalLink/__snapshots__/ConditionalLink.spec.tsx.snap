// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionalLink matches snapshot 1`] = `
<Link
  to="/some-route/index"
>
  <div>
    Sample text
  </div>
</Link>
`;
